OtripleS.Web.Api/Services/StudentService.cs

```diff
@@ -26,5 +27,20 @@ public async ValueTask<Student> DeleteStudentAsync(Guid studentId)
 
             return await this.storageBroker.DeleteStudentAsync(maybeStudent);
         }
+
+        public async ValueTask<Student> RegisterAsync(
+            Guid id,
+            string firstName,
+            string middleName,
+            string lastName,
+            DateTimeOffset BirthDate,
+            Gender gender
+        )
```

Comment:
Why not pass the entire student model?

Reply:
the Student entity will grow over time. It will not (and even now is not) apparent what data is required for the registration of a new student based on the entity alone. Having a method with explicit primitive parameters makes it obvious what the required data is.

Reply:
So if the entity grows into say 20 properties we should have methods with 20 parameters?

Reply:
the number of properties on the entity and the number of arguments required for student registration does not map 1:1, which is I've preferred this option. Yes, it is possible that in the future the number of these arguments will grow (some arguments may not even be present on the entity itself), and I agree that at some point it would become unwieldy. This is where a command/request object could come into play to replace the individual arguments with a container for them, like in the demo https://github.com/Driedas/MediatrDemo/blob/master/Mediatr.Demo/Messages/Commands/Student/RegisterStudent.cs

The alternative is to use the entity itself, that is however a tricky thing. We cannot just accept an entity, validate that it has the required minimum set of properties and persist it. When we add the Status property and others, it could be possible to register a student in a Graduated status, with a CreatedDate 10 years go and with all As across the board - his status and his academic accomplishments will most likely be referenced by navigation properties on the Student entity after all and due to persistence by reachability adopted by ORMs all of those would get persisted as well.

Reply:
While we could pick only certain properties from the Student entity to use when registering, and thus creating a new Student entity instance and persist that instead, it would not be obvious to the caller which properties those are and what the acceptable range of values are.

Reply:
For this broker-neighboring service, it doesn't really fully understand/care what a registration of a student is, it just persists a student entity in the database by calling a storage broker - there will be orchesteration, coordination, management and aggregation services downstream that will take care of some of these details you are talking about - for the time being, please consider passing the entire object, validate it's content (including the dates) and we shall run into pre-filled values versus registeration-required values eventually as the system advances - for now we are just adding entities after we verify them, that's all.

Reply:
ok, in the case that there will be an authority on top of this that will prevent persisting of nonsensical Student instances, this could work. I think that there should be some sort of check in place though that clients will not be able to call this method directly, i. e. not expose it as-is via an API.